libnm-core/nm-setting-wireless.c

```diff
@@ -774,6 +755,27 @@ verify (NMSetting *setting, NMConnection *connection, GError **error)
 		return FALSE;
 	}
 
+	if (!priv->mode || strcmp (priv->mode, NM_SETTING_WIRELESS_MODE_INFRA) == 0) {
+		if (!priv->ssid) {
+			g_set_error_literal (error,
+			                     NM_CONNECTION_ERROR,
+			                     NM_CONNECTION_ERROR_MISSING_PROPERTY,
+			                     _("property is missing"));
+			g_prefix_error (error, "%s.%s: ", NM_SETTING_WIRELESS_SETTING_NAME, NM_SETTING_WIRELESS_SSID);
+			return FALSE;
+		}
+
+		length = g_bytes_get_size (priv->ssid);
```

Comment:
the length validation is now no longer done for AP/ADHOC. That is wrong. If a SSID is present, it must always be checked for being valid.

Reply:
Good catch. Will fix.